theme/static/js/hs-vue/notifications-app.js

```diff
@@ -0,0 +1,310 @@
+                    error: function (response) {
+                        console.log(response);
+                        vue.loading = false;
+                    }
+                });
+            },
+            dismissTask: function (task) {
+                let vue = this;
+                if (vue.canBeDismissed(task)) {
+                    $.ajax({
+                        type: "GET",
+                        url: '/hsapi/_internal/dismiss_task/' + task.id,
+                        success: function (task) {
+                            vue.tasks = vue.tasks.filter(function(t) {
+                                return t.id !== task.id;
+                            });
+                        },
+                        error: function (response) {
+                            console.log(response);
+                        }
+                    });
+                }
+            },
+            canBeDismissed: function (task) {
+                return task.status === 'Completed'
+                    || task.status === 'Failed'
+                    || task.status === 'Aborted'
+                    || task.status === 'Delivered'
+            },
+            deliverTask: function(task) {
+                let vue = this;
+                if (vue.canBeDelivered(task)) {
+                    $.ajax({
+                        type: "GET",
+                        url: '/hsapi/_internal/set_task_delivered/' + task.id,
+                        success: function (task) {
+                        },
+                        error: function (response) {
+                            console.log(response);
+                        }
+                    });
+                }
+            },
+            canBeDelivered: function (task) {
+                return task.status === 'Completed'
+            },
+            abortTask: function(task) {
+                let vue = this;
+                if (vue.canBeAborted(task)) {
+                    $.ajax({
+                        type: "GET",
+                        url: '/hsapi/_internal/abort_task/' + task.id,
+                        success: function (task) {
+                            vue.registerTask(task);
+                        },
+                        error: function (response) {
+                            console.log(response);
+                        }
+                    });
+                }
+            },
+            canBeAborted: function (task) {
+                return task.name != 'resource delete' && (task.status === 'In progress'
+                    || task.status === 'Pending execution')
+            },
+            clear: function () {
+                let vue = this;
+                vue.tasks = vue.tasks.filter(function(task) {
+                    if (vue.canBeDismissed(task))
+                        vue.dismissTask(task);
+                    return !vue.canBeDismissed(task);
+                });
+            },
+            scheduleCheck() {
```

Comment:
looks like the notification app already checks the task every 1 second

Reply:
Yes, the caller does not really check the task, but asks for the notification app for the task status instead. So we are already doing task check only in one place in the notification app.